src/hooks/useOnClickOutside.ts

```diff
@@ -2,18 +2,27 @@ import { RefObject, useEffect, useRef } from 'react'
 
 export function useOnClickOutside<T extends HTMLElement>(
   node: RefObject<T | undefined>,
-  handler: undefined | (() => void)
+  handler: undefined | (() => void),
+  ignoredNodes: Array<RefObject<T | undefined>> = []
 ) {
   const handlerRef = useRef<undefined | (() => void)>(handler)
+
   useEffect(() => {
     handlerRef.current = handler
   }, [handler])
 
   useEffect(() => {
     const handleClickOutside = (e: MouseEvent) => {
-      if (node.current?.contains(e.target as Node) ?? false) {
+      const nodeClicked = node.current?.contains(e.target as Node)
```

Comment:
not blocking - but my previous comment was just to create a new variable `const clickedNode = e.target as Node` since you use it lower, and also it helps the reader understand the function a little easier